sw/device/silicon_creator/mask_rom/e2e_tests/mask_rom_e2e_testplan.hjson

```diff
@@ -0,0 +1,360 @@
+
+            - Enter bootstrap mode.
+            - Issue ERASE opcode to enter phase 2.
+            - Issue PAGE_PROGRAM opcode to program a target address.
+            - Issue ERASE opcode to the same target address.
+            - Issue RESET opcode to exit bootstrap and reset the chip.
+            - Verify that the target address was correctly erased and that the rest of flash is unchanged.
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+    // TODO: READ behavior is currently unspecified/not supported
+    {
+      name: mask_rom_e2e_phase2_read_op
+      desc: '''Verify that phase 2 of bootstrap correctly handles the READ opcode.
+
+            - Enter bootstrap mode.
+            - Issue ERASE opcode to enter phase 2.
+            - Issue PAGE_PROGRAM opcode to program a target address.
+            - Issue READ opcode to read the same target address.
+            - Verify that the READ behaves correctly.
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+
+    // Chip-specific startup
+    {
+      name: mask_rom_e2e_startup_ast_init
+      desc: '''Verify that AST is initialized on correct lifecycle states.
+
+            - For each lifecycle state:
+              - Load an OTP image corresponding to this state
+              - Boot the chip
+              - Check the SENSOR_CTRL_STATUS_AST_INIT_DONE bit
+              - This bit should be 1 if the AST should be initialized and 0 otherwise.
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+    {
+      name: mask_rom_e2e_startup_main_sram_init
+      desc: '''Verify that the main SRAM is initialized correctly after POR.
+
+            - Initiate a POR.
+            - Wait until main SRAM is initialized.
+            - Check the following main SRAM registers in SRAM_CTRL.CTRL:
+              - SCR_KEY_VALID = 1
+              - SCR_KEY_SEED_VALID = 1
+              - INIT_DONE = 1
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+    {
+      name: mask_rom_e2e_startup_retention_sram_init
+      desc: '''Verify that the retention SRAM is initialized correctly after POR.
+
+            - Initiate a POR.
+            - Wait until retention SRAM is initialized.
+            - Check the following retention SRAM registers in SRAM_CTRL.CTRL:
+              - SCR_KEY_VALID = 1
+              - SCR_KEY_SEED_VALID = 1
+              - INIT_DONE = 1
+            '''
+      target: ["verilator", "dv_sim", "fpga", "silicon"]
+      tests: []
+    }
+    {
+      name: mask_rom_e2e_startup_sram_retention
+      desc: '''Verify that retention SRAM retains information after reset.
+      
+            - Boot chip.
+            - Write known data into silicon owner reserved section of SRAM.
+            - Trigger a reset.
```

Comment:
could we discuss which reset we are talking about here? It's tracked in another bug..somewhere, but the only reset that causes retention sram to retain is low power entry / exit...do we want to include software reset in this list? There may be some hardware changes needed for that.

Reply:
I've clarified this test to be specifically low power entry/exit, not reset -- sorry for the confusion